supportedBackends/google/pipelines/common/src/test/scala/cromwell/backend/google/pipelines/common/callcaching/PipelinesApiBackendCacheHitCopyingActorSpec.scala

```diff
@@ -0,0 +1,544 @@
+      blacklistCache.hitCache.get(CallCachingEntryId(0)) shouldBe GoodCacheResult
+      blacklistCache.hitCache.get(CallCachingEntryId(1)) shouldBe BadCacheResult
+      blacklistCache.hitCache.get(CallCachingEntryId(2)) shouldBe UntestedCacheResult
+      blacklistCache.hitCache.get(CallCachingEntryId(3)) shouldBe BadCacheResult
+    }
+
+    {
+      // Step 5: a new workflow from the same grouping tries to copy an unknown cache hit from the bucket blacklisted in step 1.
+      val ioActor = TestProbe()
+      val serviceRegistryActor = TestProbe()
+      val supervisor = TestProbe()
+      val copyActor = buildCopyActor(
+        workflow = workflow2,
+        blacklistCache = blacklistCache,
+        fakeIoActor = ioActor,
+        fakeServiceRegistryActor = serviceRegistryActor,
+        supervisor = supervisor,
+        grouping = grouping)
+
+      supervisor watch copyActor
+
+      val command = buildCopyCommand(hitId = 4, bucket = LockedDownBucket)
+      copyActor ! command
+
+      supervisor.expectMsgPF(5 seconds) {
+        case _: CopyingOutputsFailedResponse =>
+      }
+      // The IoActor should not be consulted and the copying actor should simply stop itself without transitioning.
+      supervisor.expectTerminated(copyActor)
+      ioActor.expectNoMessage(5 seconds)
+
+      val List(readHit, readBucket) = instrumentationCounts(n = 2, serviceRegistryActor = serviceRegistryActor)
+
+      readHit.bucket.path.toList shouldBe expectedMetric(Hit, Read, grouping = GoogleProject, value = "4", UntestedCacheResult)
+      readBucket.bucket.path.toList shouldBe expectedMetric(Bucket, Read, grouping = GoogleProject, value = LockedDownBucket, BadCacheResult)
+
+      // Assert blacklist entries were made for bucket and hit.
+      blacklistCache.bucketCache.size() shouldBe 2
+      // The hit is bad but because the failure was generic the bucket which was previously marked good should stay that way.
+      blacklistCache.bucketCache.get(WideOpenBucket) shouldBe GoodCacheResult
+      blacklistCache.bucketCache.get(LockedDownBucket) shouldBe BadCacheResult
+
+      blacklistCache.hitCache.size() shouldBe 5
+      blacklistCache.hitCache.get(CallCachingEntryId(0)) shouldBe GoodCacheResult
+      blacklistCache.hitCache.get(CallCachingEntryId(1)) shouldBe BadCacheResult
+      blacklistCache.hitCache.get(CallCachingEntryId(2)) shouldBe UntestedCacheResult
+      blacklistCache.hitCache.get(CallCachingEntryId(3)) shouldBe BadCacheResult
+      blacklistCache.hitCache.get(CallCachingEntryId(4)) shouldBe UntestedCacheResult
+    }
+  }
+
+  private def instrumentationCounts(n: Int, serviceRegistryActor: TestProbe): List[CromwellCount] = {
+    val received = serviceRegistryActor.receiveN(n = n, max = 5 seconds).toList
+    val instrumentationCounts = received collect { case InstrumentationServiceMessage(c) => c } collect { case c: CromwellCount => c }
+    instrumentationCounts forall (c => c.value == 1 && c.sampling == 1.0)
```

Comment:
Is this line doing anything? What if the `forall` returns false?

Reply:
oops that's not right